packages/core-v2/src/semantic-model/aggregator/aggregator.ts

```diff
@@ -80,20 +95,45 @@ class SemanticModelAggregatorInternal implements SemanticModelAggregator {
     private notifyBaseModel(fromModel: SupportedModels, updated: AggregatedEntityWrapper[], removed: string[]) {
         // This does only a simple merge
         this.baseModelEntities = this.baseModelEntities
-            .filter(entity => !removed.includes(entity.id) && !updated.find(updatedEntity => updatedEntity.id === entity.id))
+            .filter(
+                (entity) =>
+                    !removed.includes(entity.id) && !updated.find((updatedEntity) => updatedEntity.id === entity.id)
+            )
             .concat(updated);
-        this.baseModelSubscribers.forEach(subscriber => subscriber(updated, removed));
+        this.baseModelSubscribers.forEach((subscriber) => subscriber(updated, removed));
     }
 
     getView(): SemanticModelAggregatorView {
         return new SemanticModelAggregatorView(this);
     }
+
+    addVisualModel(model: VisualEntityModel) {
+        if (this.visualModels.find((m) => model.getId() == m.getId())) {
+            throw new Error("Model already added.");
+        }
+
+        this.visualModels = this.visualModels.concat(model);
+        this.setActiveVisualModel(model);
+    }
+
+    deleteVisualModel(modelId: string): void {
+        this.visualModels = this.visualModels.filter((model) => model.getId() != modelId);
+        this.setActiveVisualModel(this.visualModels[0] ?? null);
+    }
+
+    setActiveVisualModel(toModel: string | VisualEntityModel | null) {
```

Comment:
Aha, já to myslel tak, že jakmile přidáš visual model, tak se stane ihned aktivním, takže nedává smysl nastavovat, které modely jsou aktivní. Tedy musel by sis někde externě držet seznam visual modelů a pak při změně ten starý odstranit a nový přidat. To ti mimo jiné umožní používat více visual modelů najednou, taky to může mít nějaký use-case.

Každopádně je to hlavně kód který budeš využívat ty, takže za mě to tak klidně nech, alespoň jde explicitněji vidět, že tam figuruje i ten vizuální model.